tests/torchtune/data/test_converters.py

```
@@ -0,0 +1,120 @@
+    "user": "Please briefly summarize this news article:\n\nAOL.com Video - Father Lets 8-Year-Old Drive On Icy Road\n\nDescription:Would you let your 8-year-old drive your car? How about on an icy road? Well one father in Russia did just that, and recorded the entire thing. To her credit, the child seemed to be doing a great job. (0:44)\n\nTags: 8-year-old driver , caught on camera , child driver , pix11\n\nSummary:",  # noqa: B950
+    "assistant": "A father in Russia allowed his 8-year-old child to drive his car on an icy road and recorded the event. The child appeared to be handling the situation well, showcasing their driving skills despite the challenging conditions.",  # noqa: B950
+}
+
+EXPECTED_MESSAGE = [
+    Message(
+        role="system",
+        content=CHAT_SAMPLE["system"],
+    ),
+    Message(
+        role="user",
+        content=CHAT_SAMPLE["user"],
+    ),
+    Message(
+        role="assistant",
+        content=CHAT_SAMPLE["assistant"],
+    ),
+]
+
+
+class TestSharegptToLlama2Messages:
+    samples = [
+        {
+            "from": "system",
+            "value": CHAT_SAMPLE["system"],
+        },
+        {
+            "from": "user",
+            "value": CHAT_SAMPLE["user"],
+        },
+        {
+            "from": "assistant",
+            "value": CHAT_SAMPLE["assistant"],
+        },
+    ]
+
+    def test_conversion(self):
+        converted_messages = sharegpt_to_llama2_messages(self.samples)
+        for converted, expected in zip(converted_messages, EXPECTED_MESSAGE):
+            assert converted == expected
+
+    def test_conversion_train_on_input(self):
+        converted_messages = sharegpt_to_llama2_messages(
+            self.samples, train_on_input=True
+        )
+        for converted, expected in zip(converted_messages, EXPECTED_MESSAGE):
+            assert converted == expected
```

do you also want to check that the mask is set properly? by default the mask is false, so you might want to set `train_on_input=False` to set mask on user message to True and properly test this functionality.

---

yes! in the next commit!